torch/autograd/__init__.py

```diff
@@ -265,7 +266,17 @@ def grad(
             cliffs. Please use ``torch._C._debug_only_display_vmap_fallback_warnings(True)``
             to show any performance warnings and file an issue on github if warnings exist
             for your use case. Defaults to ``False``.
+        materialize_grads (bool, optional): If ``True``, set the gradient for unused inputs
+            to zero instead of None. This is useful when computing higher-order derivatives.
+            If ``materialize_grads`` is ``True`` and ``allow_unused`` is ``False``, an error
+            will be raised. Defaults to ``False``.
+
     """
+    if allow_unused is False and materialize_grads is True:
```

Comment:
nit: `materialize_grads is True` -> `materialize_grads` (its a bool so we can use it directly)

Since `materialize_grads` is less likely to be False, we want to perform that check first, e.g.:
```python
materialize_grads and allow_unused is False
```
Would only need to perform one check most of the time, versus currently we'd be doing two tests everytime.

Reply:
It is indeed. Fixed in https://github.com/pytorch/pytorch/pull/97015/commits/d900d568496b1197f38dba579463526e4bdd741a